addons/mail/models/mail_followers.py

```diff
@@ -327,6 +327,8 @@ def _add_followers(self, res_model, res_ids, partner_ids, partner_subtypes, chan
                         update[fol_id] = {'subtype_ids': [(4, sid) for sid in new_sids]}
                     if fol_id and old_sids and existing_policy == 'replace':
                         update[fol_id] = {'subtype_ids': [(3, sid) for sid in old_sids]}
+                        if new_sids:
```

Comment:
Already done here above. Maybe you want to prepare a list of commands, in which case this should not be done like that.